Sherlock/sherlock.h

```diff
@@ -96,13 +99,13 @@ SOFTWARE.
 //
 // 3) `HeadUpdated()`: TBD.
 //
-// The call to `my_stream.Subscribe(my_listener);` launches the listener and returns
-// an instance of a handle, the scope of which will define the lifetime of the listener.
+// The call to `my_stream.Subscribe(my_subscriber);` launches the subscriber and returns
+// an instance of a handle, the scope of which will define the lifetime of the subscriber.
 //
-// If the subscribing thread would like the listener to run forever, it can use `.Join()` or `.Detach()`
-// on the handle. `Join()` will block the calling thread unconditionally, until the listener itself
-// decides to stop listening. `Detach()` will ensure that the ownership of the listener object
-// has been transferred to the thread running the listener, and detach this thread to run in the background.
+// If the subscribing thread would like the subscriber to run forever, it can use `.Join()` or `.Detach()`
+// on the handle. `Join()` will block the calling thread unconditionally, until the subscriber itself
+// decides to stop accepting new entries. `Detach()` will ensure that the ownership of the subscriber object
```

Comment:
The `"accepting new entries"` part is not 100% correct -- the subscriber may, for example, wait on listening to `UpdateHead()` events, instead of accepting new entries.